src/annotation_processor.md

```diff
@@ -0,0 +1,149 @@
+//<override example>
+```pdl
+@persistencePolicyInDays = 30
+record  UserVisitRecord {
+  @persistencePolicyInDays = 10
+  visitedUrls: array[URL]
+  visitedUserProfiles: array[UserRecord]
+}
+```
+Reading from the schema, a user might find the field `visitedUserProfiles`s `persistencePolicyInDays` annotation is 30 but `visitedUrls`'s `persistencePolicyInDays` annotation has value 10, which overrides the original value inherited(which was 30).
+
+
+Example case 3:  Override might also happen when some annotation needs to be updated by another value assigned from another annotation location
+```pdl
+record UserVisitRecord {
+  //...
+  @persistencePolicyInDays = 365
+  recycledChatHistories: array[chat]
+
+  @persistencePolicyInDays = 10
+  visitedUrls: array[URL]
+  //...
+}
+
+record EnterpriseUserRecord {
+  userName: UserName
+  //...
+  @persistencePolicyInDays = {"/recycledChatHistories" : 3650}
+  visitRecord: UserVisitRecord
+  //...
+}
+```
+In this example, schema `EnterpriseUserRecord` reused `UserVisitRecord` and its annotation in the field `visitRecord`, but it overrides the annotation value of `recycledChatHistories`.
+
+
+All above examples shows that inheritance and overrides gives more extensibility to the schema annotations. Users should be free to define their own rules regarding how those annotations can be read. There should be an imambigous annotation value for the fields after user's rules are applied. We call this processing step to figure out the eventual value of a field or schema's annotation "resolution" and we call such value "resolved" value.
+
+What is more, users could have their own customized logic to process the annotations and define the custom behavior when annotations is overriden or inheritted, and even have the flexibility to set a customized resolved value by calling another local function or remote procedure.  The need for such extensibility gives the motivation for Annotation Processor. We aim to build a tool that can process annotations during schema resolution, process all interested annotations by plugin-ing user's own logic and save the "resolved" value back to corresponding DataSchema.
+
+### A common application of overriding annotation using PathSpec
+In the above `EnterpriseUserRecord` example, we use `@persistencePolicyInDays = {"/recycledChatHistories" : 3650}` do denote the field we are overriding is `recycledChatHistories`. This is basically using PathSpec to reference the field or child schema that needs to be overriden. PathSpec <insert-linke-here> can be used as a path to associate relations between fields among nested schemas. Users can use PathSpec to inambigously specifcy all the paths to the child fields that they want to override.
+
+
+In Linkedin, such usage is so common that we want to use 
+we provided `PathSpecBaseVisitor`. 
```

Comment:
Introducing Visitor without explaining what it is. I suggest moving the references to PathSpecBaseVisitor to a later seciton.